dice_ml/explainer_interfaces/dice_KD.py

```diff
@@ -0,0 +1,236 @@
+"""
+Module to generate diverse counterfactual explanations based on tensorflow 2.x
+"""
+from dice_ml.explainer_interfaces.explainer_base import ExplainerBase
+import tensorflow as tf
+
+import numpy as np
+import random
+import timeit
+from sklearn.neighbors import KDTree
+import pandas as pd
+
+from dice_ml import diverse_counterfactuals as exp
+
+class DiceKD(ExplainerBase):
+
+    def __init__(self, data_interface, model_interface):
+        """Init method
+
+        :param data_interface: an interface class to access data related params.
+        :param model_interface: an interface class to access trained ML model.
+
+        """
+
+        super().__init__(data_interface) # initiating data related parameters
+
+        # initializing model variables
+        self.model = model_interface
+
+        # loading trained model
+        self.model.load_model()
+
+        # variables required to generate CFs - see generate_counterfactuals() for more info
+        self.cfs = []
+        self.features_to_vary = []
+        self.cf_init_weights = []  # total_CFs, algorithm, features_to_vary
+        self.loss_weights = []  # yloss_type, diversity_loss_type, feature_weights
```

Comment:
do you need the `self.loss weights` ? try to include only those that are needed later.

Reply:
same for `optimizer_weights`